crypto/s2n_libcrypto.c

```diff
@@ -153,6 +153,18 @@ S2N_RESULT s2n_libcrypto_init(void)
     return S2N_RESULT_OK;
 }
 
+#if S2N_OPENSSL_VERSION_AT_LEAST(3, 0, 0)
+int s2n_libcrypto_teardown_cb(OSSL_PROVIDER *provider, void *cbdata) {
+    return OSSL_PROVIDER_unload(provider);
+}
+#endif
+
+void s2n_libcrypto_teardown(void) {
+#if S2N_OPENSSL_VERSION_AT_LEAST(3, 0, 0)
+    OSSL_PROVIDER_do_all(NULL, *s2n_libcrypto_teardown_cb, NULL);
+#endif
+}
```

Comment:
It looks like `OSSL_PROVIDER_do_all` returns 1 for success and 0 for error. Would it make sense to wrap this with RESULT_GUARD_OSSL and return an S2N_CLEANUP_RESULT instead of void, just in case this fails for some reason?